vagrant/environments/foss/config.yaml

```diff
@@ -3,7 +3,7 @@
 vm:
   memory: 1024                # MB or RAM to assign
   cpu: 1                      # Number of vCPU to assign to the VM
-  box: ubuntu1604             # Box used for the VM, from the box list in vagrant/boxes.yaml
+  box: centos7             # Box used for the VM, from the box list in vagrant/boxes.yaml
```

Comment:
If you want only the master on Centos, you should add 
box: centos7 
only for the Puppet node, and leave the default ubuntu1604 here